models/enterprise_sonic/vlan_mapping/sonic_vlan_mapping.yml

```diff
@@ -0,0 +1,94 @@
+---
+GENERATOR_VERSION: '1.0'
+
+ANSIBLE_METADATA: |
+    {
+      'metadata_version': '1.1',
+      'status': ['preview'],
+      'supported_by': 'community',
+      'license': 'Apache 2.0'
+    }
+NETWORK_OS: sonic
+RESOURCE: vlan_mapping
+COPYRIGHT: Copyright 2023 Dell Inc. or its subsidiaries. All Rights Reserved
+
+DOCUMENTATION: |
+  module: sonic_vlan_mapping
+  author: "Cypher Miller (@Cypher-Miller)"
+  version_added: "2.1.0"
+  short_description: Configure vlan mappings on SONiC.
+  description:
+    - This module provides configuration management for vlan mappings on devices running SONiC.
+    - Vlan mappings only available on TD3 and TD4 devices.
+    - For TD4 devices must enable vlan mapping first (can enable in config-switch-resource).
+  options:
+    config:
+      description:
+        - Specifies the vlan mapping related configurations.
+      type: list
+      elements: dict
+      suboptions:
+        name:
+          description:
+            - Full name of the interface, i.e. Ethernet8, PortChannel2, Eth1/2.
+          required: true
+          type: str
+        mapping:
+          description:
+            - Defining a single vlan mapping.
+          type: list
+          elements: dict
+          suboptions:
+            service_vlan:
+              description:
+                - Configure service provider VLAN ID. 
+                - VLAN ID range is 1-4094.
+              required: true
+              type: int
+            vlan_ids:
+              description:
+                - Configure customer VLAN IDs.
+                - If mode is double tagged translation then this VLAN ID represents the outer VLAN ID.
+                - If mode is set to stacking can pass ranges and/or multiple list entries.
+              type: list
```

Comment:
Instead of nesting the vlan_ids in a list of dicts, each of which can contain just a string, it would be simpler for the user and easier to implement if the elements of the top level "vlan_ids" list were of type "str".

i.e.
```
elements: str
```

 The additional instructional information currently provided in the inner "vlan_ids" description can be appended to the top level "vlan_ids" description.

Reply:
Fixed this in the new commits.